aten/src/ATen/native/Unique.cpp

```diff
@@ -0,0 +1,45 @@
+// Returns unique elements of input tensor.
+
+#include <unordered_map>
+#include <unordered_set>
+
+#include "ATen/ATen.h"
+
+namespace at {
+namespace native{
+
+std::tuple<Tensor, Tensor> unique(
+    const Tensor& self, const bool sorted, const bool return_inverse) {
+  std::unordered_set<int64_t> set(
+      self.data<int64_t>(), self.data<int64_t>() + self.numel());
+  Tensor output = self.type().tensor({static_cast<long long>(set.size())});
+  
+  if (!sorted) {
+    std::copy(set.begin(), set.end(), output.data<int64_t>());
+  } else {
+    std::vector<int64_t> vec(set.begin(), set.end());
+    std::sort(vec.begin(), vec.end());
+    std::copy(vec.begin(), vec.end(), output.data<int64_t>());
+  }
+
+  Tensor inverse_indices;
+  if (!return_inverse) {
+    inverse_indices = self.type().toScalarType(kLong).tensor({0});
+    
+  } else {
+    inverse_indices = self.type().toScalarType(kLong).tensor(self.sizes());    
```

Comment:
I think you want `at::empty_like(self)` or `at::zero_like(self)`. That will create a tensor with the same shape and type

Reply:
For the inverse indices, though, I think it always make sense for them to be of type kLong regardless of what the input tensor type is (assuming support for non-LongTensor input is added) - I just want them to have the same shape as the input

Reply:
Then maybe `at::CPU(kLong).tensor(self.sizes())`

Reply:
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/README.md#can-it-handle-being-passed-variables specifically calls out to never use at::CPU to deal with inheritance issues - is that not relevant in this case?

Reply:
No you're right, I wasn't aware of this. Let's leave it then.